setup.py

```diff
@@ -117,140 +118,81 @@
+        "windows.txt",
+    )
+]
+SALT_LONG_DESCRIPTION_FILE = os.path.join(os.path.abspath(SETUP_DIRNAME), "README.rst")
 
 # pylint: disable=W0122
 exec(compile(open(SALT_VERSION).read(), SALT_VERSION, "exec"))
 # pylint: enable=W0122
 
 
 # ----- Helper Functions -------------------------------------------------------------------------------------------->
-
-
-def _parse_op(op):
-    """
-    >>> _parse_op('>')
-    'gt'
-    >>> _parse_op('>=')
-    'ge'
-    >>> _parse_op('=>')
-    'ge'
-    >>> _parse_op('=> ')
-    'ge'
-    >>> _parse_op('<')
-    'lt'
-    >>> _parse_op('<=')
-    'le'
-    >>> _parse_op('==')
-    'eq'
-    >>> _parse_op(' <= ')
-    'le'
-    """
-    op = op.strip()
-    if ">" in op:
-        if "=" in op:
-            return "ge"
-        else:
-            return "gt"
-    elif "<" in op:
-        if "=" in op:
-            return "le"
-        else:
-            return "lt"
-    elif "!" in op:
-        return "ne"
-    else:
-        return "eq"
-
-
-def _parse_ver(ver):
-    """
-    >>> _parse_ver("'3.4'  # pyzmq 17.1.0 stopped building wheels for python3.4")
-    '3.4'
-    >>> _parse_ver('"3.4"')
-    '3.4'
-    >>> _parse_ver('"2.6.17"')
-    '2.6.17'
-    """
-    if "#" in ver:
-        ver, _ = ver.split("#", 1)
-        ver = ver.strip()
-    return ver.strip("'").strip('"')
-
-
-def _check_ver(pyver, op, wanted):
-    """
-    >>> _check_ver('2.7.15', 'gt', '2.7')
-    True
-    >>> _check_ver('2.7.15', 'gt', '2.7.15')
-    False
-    >>> _check_ver('2.7.15', 'ge', '2.7.15')
-    True
-    >>> _check_ver('2.7.15', 'eq', '2.7.15')
-    True
-    """
-    pyver = distutils.version.LooseVersion(pyver)
-    wanted = distutils.version.LooseVersion(wanted)
-    if IS_PY3:
-        if not isinstance(pyver, str):
-            pyver = str(pyver)
-        if not isinstance(wanted, str):
-            wanted = str(wanted)
-    return getattr(operator, "__{}__".format(op))(pyver, wanted)
-
-
 def _parse_requirements_file(requirements_file):
-    parsed_requirements = []
+    # Since there are no ordered set's in python, let's use an ordered dictionary
+    parsed_requirements = collections.OrderedDict()
     with open(requirements_file) as rfh:
         for line in rfh.readlines():
             line = line.strip()
-            if not line or line.startswith(("#", "-r")):
+            if not line or line.startswith("#"):
                 continue
-            if IS_WINDOWS_PLATFORM:
-                if "libcloud" in line:
-                    continue
-            if IS_PY3 and "futures" in line.lower():
-                # Python 3 already has futures, installing it will only break
-                # the current python installation whenever futures is imported
+            if line.startswith("-r"):
+                include_requirements_file = os.path.join(
+                    os.path.dirname(requirements_file),
+                    line.split("-r")[1].split("#")[0].strip(),
```

Comment:
I'm not sure, but isn't there a chance to have a file name with spaces? Maybe `line.split(' ', 1)`?